GoBoard.java

```diff
@@ -0,0 +1,139 @@
+                board[row][col] = new Square(row, col);
+                super.add(board[row][col]);
+            }
+        }
+        repaint();
+    }
+
+	// a method to place the stones in the intersections using a mouse click and paints/draw the lines on the board
+    private class Square extends JPanel {
+
+        Stone stone;
+        final int row;
+        final int column;
+
+        Square(int r, int c) {
+            stone = Stone.NONE;
+            row = r;
+            column = c;
+            super.addMouseListener(new MouseAdapter(){
+                @Override
+                public void mouseClicked(MouseEvent me) {
+                    if(stone != Stone.NONE) return;
+                    stone = whiteToMove ? Stone.WHITE : Stone.BLACK;
+                    whiteToMove = !whiteToMove;
+                    repaint();
+                }
+            });
+        }
+
+ 	// a method that paints/draws the lines of the board and sets its color. 
+        protected void paintComponent(Graphics g) {
+
+            // call paintComponent to ensure the panel displays correctly
+            super.paintComponent(g); 
+
+            int width = super.getWidth(); // total width
+            int height = super.getHeight(); // total height
+
+            g.setColor(new Color(0xB78600));// set the background color
+
+            g.fillRect(0, 0, width, height);// the background color will fill the whole board
+
+            g.setColor(Color.BLACK);// set color for the vertical and horizontal lines of the board 
+
+            if(row == 0 || row == board.length-1 || column == 0 || column == board.length-1) { // draw the lines of the board
+                if(column == 0) {
+                    g.drawLine(width/2, height/2, width, height/2);
+                    if(row == 0) g.drawLine(width/2, height/2, width/2, height);
+                    else if(row == 18) g.drawLine(width/2, height/2, width/2, 0);
+                    else g.drawLine(width/2, 0, width/2, height);
+                }
+                else if(column == 18) {
+                    g.drawLine(0, height/2, width/2, height/2);
+                    if(row == 0) g.drawLine(width/2, height/2, width/2, height);
+                    else if(row == 18) g.drawLine(width/2, height/2, width/2, 0);
+                    else g.drawLine(width/2, 0, width/2, height);
+                }
+                else if(row == 0) {
+                    g.drawLine(0, height/2, width, height/2);
+                    g.drawLine(width/2, height/2, width/2, height);
+                }
+                else {
+                    g.drawLine(0, height/2, width, height/2);
+                    g.drawLine(width/2, height/2, width/2, 0);
+                }
+            } else {
+                g.drawLine(0, height/2, width, height/2);
+                g.drawLine(width/2, 0, width/2, height);
+            }
+            stone.paint(g, width);
+        }
+    }
+}
+
+
+enum Stone { 
```

Comment:
Having an enum with methods is very clunky and in bad style, I didn't even think that Java allowed this.